### ISSUE TITLE
[Prometheus Exporter] Unable to disable `otel_scope_name` attribute

### ISSUE THREAD

**Issue by timwoj** [open-telemetry/opentelemetry-cpp#2442]
According to the spec for the Prometheus exporter: 

> Prometheus exporters SHOULD provide a configuration option to disable the otel_scope_info metric and otel_scope_ labels.

I can't find any way to actually disable these. The prometheus exporter would disable them assuming the strings passed in via the scope data are empty. The data in the `InstrumentationScope` that gets passed into the exporter comes from agruments passed to `GetMeter()`. Passing a blank string for `library_name` would ostensibly disable it based on the code in the exporter, but passing a blank there results in an error:

```
[Warning] File: <>/auxil/opentelemetry-cpp/sdk/src/metrics/meter_provider.cc:54 [MeterProvider::GetMeter] Library name is empty.
```

**Comment by timwoj** [comment]
> See `PrometheusExporterOptions::populate_target_info`.
> 
> This can be set programatically.

I set this to false and `otel_scope_name` is still part of every instrument in the output. It looks like all this setting does is remove the `target_info` instrument from being reported.

**Comment by marcalff** [comment]
Indeed, `populate_target_info` is for something different, sorry about the confusion.

So we need a new exporter option attributes, and the environment variable that goes with it, this is a valid issue.

**Comment by marcalff** [comment]
Reopen.

The collide with the following spec PR, which defines the option and environment variable to use:

*  https://github.com/open-telemetry/opentelemetry-specification/pull/3796

**Comment by timwoj** [comment]
Looks like the spec PR was merged. I'll try to get a PR that renames it done this week.

**Comment by timwoj** [comment]
> Looks like the spec PR was merged. I'll try to get a PR that renames it done this week.

Should I rename all of the `populate_otel-_scope` variables in the original PR, or just the environment variable? I'm only asking because of the clash with the naming of `populate__target_info`.

**Comment by timwoj** [comment]
Ping on the above question, which probably got lost in the holidays.

**Comment by marcalff** [comment]
@timwoj 

Everything related to `populate_otel_scope` is not published yet in opentelemetry-cpp, because it was merged after the last release 1.13.0.

So, please rename C++ member names, parameter names, etc, and the environment variable name.

Once this renaming is done, we can then ship a clean 1.14.0.

Not sure what conflict with `populate_target_info` this can cause, if any.

Given that the spec itself is in experimental status for this area, renaming can be done around `populate_target_info` if necessary, we just have to document this as a breaking change in the changelog (breaking changes in the SDK are acceptable).
